api/src/test/java/com/epam/pipeline/acl/docker/ToolGroupApiServiceTest.java

```diff
@@ -0,0 +1,412 @@
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        doReturn(toolGroup).when(mockToolGroupManager).create(toolGroup);
+
+        assertThat(toolGroupApiService.create(toolGroup)).isEqualTo(toolGroup);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyCreateToolGroupWithInvalidRole() {
+        initAclEntity(dockerRegistry, AclPermission.WRITE);
+        doReturn(toolGroup).when(mockToolGroupManager).create(toolGroup);
+
+        assertThrows(AccessDeniedException.class, () -> toolGroupApiService.create(toolGroup));
+    }
+
+    @Test
+    @WithMockUser(roles = TOOL_GROUP_MANAGER)
+    public void shouldDenyCreateToolGroupWhenPermissionIsNotGranted() {
+        initAclEntity(dockerRegistry);
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        doReturn(toolGroup).when(mockToolGroupManager).create(toolGroup);
+
+        assertThrows(AccessDeniedException.class, () -> toolGroupApiService.create(toolGroup));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldUpdateToolGroupForAdmin() {
+        doReturn(toolGroup).when(mockToolGroupManager).updateToolGroup(toolGroup);
+
+        assertThat(toolGroupApiService.update(toolGroup)).isEqualTo(toolGroup);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldUpdateToolGroupWhenPermissionIsGranted() {
+        initAclEntity(toolGroup, AclPermission.WRITE);
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        doReturn(toolGroup).when(mockToolGroupManager).updateToolGroup(toolGroup);
+
+        assertThat(toolGroupApiService.update(toolGroup)).isEqualTo(toolGroup);
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDenyUpdateWhenPermissionIsNotGranted() {
+        initAclEntity(toolGroup);
+
+        assertThrows(AccessDeniedException.class, () -> toolGroupApiService.update(toolGroup));
+    }
+
+    @Test
+    @WithMockUser(roles = ADMIN_ROLE)
+    public void shouldLoadToolGroupsByRegistryIdForAdmin() {
+        final List<ToolGroup> toolGroups = mutableListOf(toolGroup);
+        doReturn(toolGroups).when(mockToolGroupManager).loadByRegistryId(ID);
+
+        assertThat(toolGroupApiService.loadByRegistryId(ID)).isEqualTo(toolGroups);
+    }
+
+    @Test
+    @WithMockUser(username = SIMPLE_USER)
+    public void shouldLoadToolGroupsByRegistryIdWhenPermissionIsGranted() {
+        initAclEntity(anotherToolGroup, AclPermission.READ);
+        doReturn(SIMPLE_USER).when(mockAuthManager).getAuthorizedUser();
+        final List<ToolGroup> toolGroups = mutableListOf(anotherToolGroup);
+        doReturn(toolGroups).when(mockToolGroupManager).loadByRegistryId(ID);
+
+        final List<ToolGroup> returnedTollGroups = toolGroupApiService.loadByRegistryId(ID);
+
+        assertThat(returnedTollGroups).isEqualTo(toolGroups);
+        assertThat(returnedTollGroups.get(0).getMask()).isEqualTo(1);
```

Comment:
Maybe  you can replace `1` with something like `private static final int READ_PERMISSION` for better readability (as I did in the tests on `DataStorageApiService`). Or, in case you only check the `READ` permission mask, you can replace `1` with `AclPermission.READ.getMask()` (as it done in the Gitlab project). The same at line 195.